scripts/merge-fork/replace-fork.js

```diff
@@ -18,31 +19,68 @@ const argv = minimist(process.argv.slice(2), {
 });
 
 async function main() {
+  const status = execSync('git status --porcelain').toString();
+  const hadUnstagedChanges = status.includes('?? ') || status.includes(' M ');
+  if (hadUnstagedChanges) {
+    const readline = require('readline');
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+
+    await new Promise(resolve => {
+      rl.question(
+        `\n${chalk.yellow.bold(
+          'Unstaged changes were found in repository.'
+        )} Do you want to continue? (Y/n) `,
+        input => {
+          switch (input.trim().toLowerCase()) {
+            case '':
+            case 'y':
+            case 'yes':
+              resolve();
+              break;
+            default:
+              console.log('No modifications were made.');
+              process.exit(0);
+              break;
+          }
+        }
+      );
+    });
+  }
+
   const oldFilenames = await glob('packages/react-reconciler/**/*.old.js');
   await Promise.all(oldFilenames.map(unforkFile));
 
   // Use ESLint to autofix imports
-  spawnSync('yarn', ['linc', '--fix'], {
+  const spawn = spawnSync('yarn', ['linc', '--fix'], {
     stdio: ['inherit', 'inherit', 'pipe'],
   });
-  // TODO: If eslint crashes, it may not have successfully fixed all
-  // the imports, which would leave the reconciler files in an inconsistent
-  // state. So we used to crash and reset the working directory. But that
-  // solution assumed that the working directory was clean before you run the
-  // command — if it wasn't, it'll not only reset the synced reconciler files,
-  // but all the other uncommitted changes.
-  //
-  // We need a different strategy to prevent loss of work. For example, we could
-  // exit early if the working directory is not clean before you run the script.
-  //
-  // Until we think of something better, I've commmented out this branch to
-  // prevent work from accidentally being lost.
-  // if (spawn.stderr.toString() !== '') {
-  //   spawnSync('git', ['checkout', '.']);
-
-  //   console.log(Error(spawn.stderr));
-  //   process.exitCode = 1;
-  // }
+
+  if (spawn.stderr.toString() !== '') {
```

Comment:
I think this also passes for a regular ESLint error instead of a crash. Could only run this branch if the command actually crashes? If it's only an error (like an unused variable, say) then the import fixes will still have worked.

Reply:
Ah, good call. I should be using `spawn.code === 1` to check for this instead.